react/src/components/UserResourcePolicySettingModal.tsx

```diff
@@ -1,64 +1,85 @@
 import { GBToBytes, bytesToGB } from '../helper';
+import { useSuspendedBackendaiClient } from '../hooks';
 import BAIModal, { BAIModalProps } from './BAIModal';
+import Flex from './Flex';
+import FormItemWithUnlimited from './FormItemWithUnlimited';
+import {
+  CreateUserResourcePolicyInput,
+  UserResourcePolicySettingModalCreateMutation,
+} from './__generated__/UserResourcePolicySettingModalCreateMutation.graphql';
 import { UserResourcePolicySettingModalFragment$key } from './__generated__/UserResourcePolicySettingModalFragment.graphql';
-// import { UserResourcePolicySettingModalCreateMutation } from "./__generated__/UserResourcePolicySettingModalCreateMutation.graphql";
-import { UserResourcePolicySettingModalModifyMutation } from './__generated__/UserResourcePolicySettingModalModifyMutation.graphql';
-import { Form, Input, message, Alert, FormInstance } from 'antd';
+import {
+  ModifyUserResourcePolicyInput,
+  UserResourcePolicySettingModalModifyMutation,
+} from './__generated__/UserResourcePolicySettingModalModifyMutation.graphql';
+import { Form, Input, Alert, App, InputNumber, theme } from 'antd';
 import graphql from 'babel-plugin-relay/macro';
-import React, { useRef } from 'react';
+import _ from 'lodash';
+import React, { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useFragment, useMutation } from 'react-relay';
 
 interface Props extends BAIModalProps {
-  // userResourcePolicy: string;
+  existingPolicyNames?: string[];
   userResourcePolicyFrgmt: UserResourcePolicySettingModalFragment$key | null;
-  onRequestClose: () => void;
+  onRequestClose: (success?: boolean) => void;
 }
 
 const UserResourcePolicySettingModal: React.FC<Props> = ({
-  userResourcePolicyFrgmt: resourcePolicyFrgmt,
+  existingPolicyNames,
+  userResourcePolicyFrgmt = null,
   onRequestClose,
   ...baiModalProps
 }) => {
   const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { message } = App.useApp();
+  const [form] = Form.useForm();
 
-  const formRef = useRef<FormInstance>(null);
+  const baiClient = useSuspendedBackendaiClient();
+  const supportMaxVfolderCount = baiClient?.supports(
+    'max-vfolder-count-in-user-and-project-resource-policy',
+  );
+  const supportMaxQuotaScopeSize = baiClient?.supports('max-quota-scope-size');
+  const supportMaxSessionCountPerModelSession = baiClient?.supports(
+    'max-session-count-per-model-session',
+  );
+  const supportMaxCustomizedImageCount = baiClient?.supports(
+    'max-customized-image-count',
+  );
 
-  const userResourcePolicyInfo = useFragment(
+  const userResourcePolicy = useFragment(
     graphql`
       fragment UserResourcePolicySettingModalFragment on UserResourcePolicy {
         id
         name
-        created_at
-        # max_vfolder_count
-        max_quota_scope_size
+        max_vfolder_count @since(version: "23.09.6")
```

Comment:
According to `schema.graphql`, `max_vfolder_count` of `UserResourcePolicy` is added since 24.03.1